frc/1. Basics/5a. First Subsystem (Roller Intake).md

```diff
@@ -1,7 +1,200 @@
 # First Subsystem(Roller Intake)
 
-Now that we have gone over what Command-Based is and what actuators are, we can now try programming our first subsystem: a Roller Intake.
+Now that we have gone over what the command-based paradigm is and what actuators are, we can now try programming our first subsystem: a roller intake.
 Before we start programming it, it is important to understand what it actually is.
 
+## Subsystem Description
 
+![intake](img/rollerintake.jpg)
 
+The image above is the intake of 1257's robot for the 2019 game Destination Deep Space. The wheels on the outside of the intake roll inwards to take in balls. It can then eject the balls by spinning outward. While the ball is within the intake, we will keep it in place by not spinning the wheels at all.
+
+These wheels were moved by a motors which were controlled by motor controllers (the things we actually program). For the purposes of this tuturial, we will assume that SPARK MAXes were used. Now that we have the subsystem basics out of the way, it's time to start coding!
+
+## Snail Subsystem
+
+Before we make our file that represents the subsystem in code, there is one file in our template which is essential: `SnailSubsystem.java`. This file is essentially an abstract class that each subsystem class extends in order to gain key abstract functions. For a refresh on what abstract classes are and their uses go to our [Polymorphism page](https://github.com/FRC1257/robotics-training/blob/Ethan-Subsystem/java/4.%20Objects/6.%20Polymorphism.md).
+
+Here is a copy of the SnailSubsystem code.
+
+```java
+
+package frc.robot.subsystems;
+
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+
+public abstract class SnailSubsystem extends SubsystemBase {
+
+    public abstract void update();
+    public abstract void outputValues();
+    public abstract void setUpConstantTuning();
+    public abstract void getConstantTuning();
+}
+```
+
+**Now for the line by line review:**

```

Comment:
For this part the line by line is fine but when you actually get into programming a subsystem which people should be understanding more deeply so they can do it for their own subsystems, imo this style is bad